File: python/test/test_transform_reduce.py
import ThrustRTC as trtc



darr = trtc.device_vector_from_list([ -1, 0, -2, -2, 1, -3], 'int32_t')

absolute_value = trtc.Functor( {}, ['x'], 
'''
         return x<(decltype(x))0 ? -x : x;
''')

print(trtc.Transform_Reduce(darr, absolute_value, trtc.DVInt32(0), trtc.Maximum()))

File: java/test/test_transform_reduce.java
import java.util.*;
import JThrustRTC.*;

public class test_transform_reduce
{
	public static void main(String[] args) 
	{
        Functor absolute_value = new Functor ( new String[]{ "x" }, "        return x<(decltype(x))0 ? -x : x;\n" );
        DVVector d_data = new DVVector(new int[] { -1, 0, -2, -2, 1, -3 });
        System.out.println(TRTC.Transform_Reduce(d_data, absolute_value, new DVInt32(0), new Functor("Maximum")));
	}
}
